lib/courier.ex

```diff
@@ -33,12 +36,21 @@ defmodule Courier do
   end
 
   def start_link(mailer, config) do
-    GenServer.start_link(__MODULE__, [mailer, config])
+    Supervisor.start_link(__MODULE__, [mailer, config])
   end
 
   def init([mailer, config]) do
-    _ = mailer.__adapter__.init(config)
-    {:ok, mailer}
+    import Supervisor.Spec
+
+    adapter = mailer.__adapter__
+    task_sup = Module.concat(mailer, TaskSupervisor)
+    config = Keyword.put(config, :task_sup, task_sup)
+
+    children = [
+      worker(adapter, [config]),
+      supervisor(Task.Supervisor, [[name: task_sup]])
```

Comment:
This is the Task Supervisor start_link I was referring to. Do we need the one that is in init?